main.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+type node struct {
+	val         int
+	left, right *node
+}
+
+func levelOrderTraversal(root *node, sink func(v int)) (res []int) {
+	level := make(map[int][]int)
+	var visit func(root *node, level int, levelMapNode *map[int][]int)
```

Comment:
Maps are passed by a reference to a function, so a pointer to the map is not needed.